INTEGRATION.md

```diff
@@ -44,34 +77,37 @@ The below decision tree defines the decisions to be made in order to choose the
+
+\** `try racing gateway` depends on the environment variable `IPFS_RACING_GATEWAY`. See below for details.
 
-\** `find the best* gateway`. The heuristics to find the best gateway are as follows. A racing gateway logic fires of a request to n-th gateways simultaneously (say 10 gateways out of a list of potentially 100's). Of those 10, the one that responds fastest is stored in a list. If this one is already responding within 50ms then this gateway is used. If this one is taking more then 200ms then the next batch of 10 gateways is probed. This flow continues till:
-1. a gateway with a response below 50ms has been found
-2. gateways have been probed for over 2 seconds (in which case it just stops and uses the fastest one).
+\*** `find the best* gateways`. The heuristics to find the best gateway are as follows. A racing gateway logic fires a request to n-th gateways simultaneously (say 10 gateways out of a list of potentially 100's). Of those 10, those that respond within 300ms are appended in $GWFILE. This flow continues till:
+1. all gateways in the list have been checked
+2. more then 2 seconds have passed in which case it aborts all current and pending requests
 
-The result of this probe will be stored in `$CONFIG/ipfs/racing_gateway_response` as a single line being the gateway that responded fastest. In subsequent racing gateway requests this file will be read and used as starting point. If this gateway still responds within 50ms then no other gateways will be probed.
+The result of this probe will be stored in `$GWFILE` as a list of gateways that responded within the set bounds.
 
-\*** `dweb.link`, see the below `IPFS_FALLBACK_GATEWAY` for details.
+\**** `dweb.link`, see the below `IPFS_FALLBACK_GATEWAY` for details.
 
 #### Environment variables
 
-The decision tree is influences by a couple environement variables.
+The decision tree is influenced by a couple environement variables.
 
-**`IPFS_RACING_GATEWAY`** When this environemnt variable isn't found (the default), racing gateways should be attempted upon reaching this point in the flow. When the environment variable exists it's value should be used. If it's value is `1` (or `true`) then racing gateways should be attempted. If this value is `0` (or `false`) then racing gateways are off. The control flow proceeds in the `no` branch.
+**`IPFS_RACING_GATEWAY`** When this environemnt variable isn't found (the default), racing gateways will be attempted upon reaching this point in the flow. When the environment variable exists it's value should be used. If it's value is `1` (or `true`) then racing gateways should be attempted. If this value is `0` (or `false`) then racing gateways are off. The control flow proceeds in the `no` branch.
 
 **`IPFS_FALLBACK_GATEWAY`** When this variable doesn't exist `dweb.link` will be used. When the variable does exist it's value will be used instead.
```

Comment:
I don't think this belongs to IPIP defining `gateways` file.
These concerns should be moved to separate one, which defines the consumer of the `gateways` file. 

Perhaps:

- Move this spec to `./integrations/GATEWAYS_FILE.md` for now
- Open separate IPIP and add `./integrations/GATEWAYS_CLIENT.md`